v2/sourcedb-to-spanner/src/main/java/com/google/cloud/teleport/v2/source/reader/io/jdbc/iowrapper/JdbcIoWrapper.java

```diff
@@ -287,12 +289,23 @@ private static TableConfig getTableConfig(
      */
     if (indexInfo.containsKey(tableName)) {
       ImmutableList<SourceColumnIndexInfo> tableIndexInfo = indexInfo.get(tableName);
-
       // TODO(vardhanvthigle): support for non-primary indexes.
-      tableIndexInfo.stream()
-          .filter(info -> info.isPrimary() && info.ordinalPosition() == 1)
-          .map(SourceColumnIndexInfo::cardinality)
-          .forEach(tableConfigBuilder::setApproxRowCount);
+      Optional<Long> tableCardinality =
+          tableIndexInfo.stream()
+              .filter(info -> info.isPrimary() && info.ordinalPosition() == 1)
+              .map(SourceColumnIndexInfo::cardinality)
+              .findFirst();
+
+      tableCardinality.ifPresentOrElse(
```

Comment:
Although the error is about cardinality, root cause is having no supported primary key.
I am not sure if we need to check of cardinality to throw no Pkey Error.

[nit, java-syle] 
In case you would need it, Consider using `noneMatch` for throwing Exception through the stream itself.

Reply:
```
Although the error is about cardinality, root cause is having no supported primary key.
I am not sure if we need to check of cardinality to throw no Pkey Error.
```

Cardinality check is how the logic was implemented previously, so I have preserved that - 

```
.filter(info -> info.isPrimary() && info.ordinalPosition() == 1)
          .map(SourceColumnIndexInfo::cardinality)
          .forEach(tableConfigBuilder::setApproxRowCount);
```

```
[nit, java-syle]
In case you would need it, Consider using noneMatch for throwing Exception through the stream itself.
```

I am aware of that but I find the Optional approach more readable in nature.

Reply:
```
Cardinality check is how the logic was implemented previously, so I have preserved that -
```
For Cardinality not found we already have an error.
If we are checking for Cardinality outside the natural place where that error originates, we might aswell reword that error and achieve the same result.

The root cause of that error happens to be no suitable Pkey .

Reply:
Please provide me with an code snippet/example of what you have in mind and I would be happy to change to that :)